spring/spring5/spring5-common/src/test/java/org/infinispan/spring/common/session/util/EventsWaiter.java

```diff
@@ -29,4 +35,30 @@ public static void assertNumberOfEvents(Supplier<Stream<ApplicationEvent>> event
 
    }
 
+   public static void assertSessionContent(Supplier<Stream<ApplicationEvent>> eventCollector,
+                                           Class<? extends AbstractSessionEvent> eventClass,
+                                           String sessionId,
+                                           String attName,
+                                           String attValue,
+                                           int timeout,
+                                           TimeUnit timeoutUnit) {
+
+      long stopTime = System.currentTimeMillis() + timeoutUnit.toMillis(timeout);
+
+      List<Session> sessions = null;
+
+      while (System.currentTimeMillis() < stopTime) {
```

Comment:
You need to modify the loop condition or add a break, otherwise the test will always block for 2 seconds here

Reply:
I did something else